src/main/java/ezvcard/io/json/JCardRawReader.java

```diff
@@ -86,11 +99,13 @@ public void readNext(JCardDataStreamListener listener) throws IOException {
 		this.listener = listener;
 
 		//find the next vCard object
-		JsonToken prev = null;
+		JsonToken prev = parser.getCurrentToken();
 		JsonToken cur;
 		while ((cur = parser.nextToken()) != null) {
 			if (prev == JsonToken.START_ARRAY && cur == JsonToken.VALUE_STRING && "vcard".equals(parser.getValueAsString())) {
 				break;
+			} else if (strict) {
+				throw new JCardParseException(JsonToken.VALUE_STRING, cur);
```

Comment:
This is not properly account for the case when the value is a string, but it's != "vcard"